rust/src/lib.rs

```diff
@@ -1375,33 +1518,45 @@ pub extern "C" fn zcashlc_shield_funds(
             slice::from_raw_parts(output_params, output_params_len)
         }));
 
-        //grab secret private key for t-funds
-        let sk:SecretKey = (&Wif(tsk_wif.to_string())).try_into()?;
+        // decode secret private key for spending t-funds
+        let t_account_privkey = hex::decode(tsk)
+            .map_err(|_| format_err!("error decoding hex string"))
+            .and_then(|tsk_bytes| {
+                ExtendedPrivKey::deserialize(&tsk_bytes)
+                    .map_err(|_| format_err!("error decoding transparent spending key."))
+            })
+            .map(|decoded_account_key| {
+                AccountPrivKey::from_extended_privkey(decoded_account_key)
+            })
+            .unwrap();
 
-        let extsk =
+        // decode the Sapling extfvk corresponding to the account to which we are
+        // shielding the funds
+        let extfvk =
             match decode_extended_spending_key(network.hrp_sapling_extended_spending_key(), &extsk)
             {
-                Ok(Some(extsk)) => extsk,
+                Ok(Some(extsk)) => ExtendedFullViewingKey::from(&extsk),
                 Ok(None) => {
                     return Err(format_err!("ExtendedSpendingKey is for the wrong network"));
-                },
+                }
                 Err(e) => {
                     return Err(format_err!("Invalid ExtendedSpendingKey: {}", e));
-                },
+                }
             };
-        
+
         let memo = Memo::from_str(&memo).map_err(|_| format_err!("Invalid memo"))?;
         let memo_bytes = MemoBytes::from(memo);
-        // shield_funds(&db_cache, &db_data, account, &tsk, &extsk, &memo, &spend_params, &output_params)
-        shield_funds(&mut update_ops, 
-            &network, 
-            LocalTxProver::new(spend_params, output_params), 
-            AccountId(account), 
-            &sk,
-            &extsk, 
-            &memo_bytes, 
-            ANCHOR_OFFSET) 
-            .map_err(|e| format_err!("Error while shielding transaction: {}", e))
+        shield_transparent_funds(
+            &mut update_ops,
+            &network,
+            LocalTxProver::new(spend_params, output_params),
+            &t_account_privkey,
+            &extfvk,
+            AccountId::from(account),
+            &memo_bytes,
+            ANCHOR_OFFSET,
```

Comment:
Dropping this here so I don't forget: I noticed that the Android SDK uses 0 here, which seems wrong (that would try and spend mempool funds).

Reply:
The android SDK uses a version of the rust code that has a bug where funds took 10 confirmations or more. That's why you see a zero there. A problem that the iOS SDK also had and it's documented in many issues I think on the zcash-ios-wallet repo. Also another one of them is this one

https://github.com/zcash-hackworks/zcash-light-client-ffi/issues/8